homeassistant/components/sensor/statistics.py

```diff
@@ -143,26 +142,22 @@ def should_poll(self):
     def device_state_attributes(self):
         """Return the state attributes of the sensor."""
         if not self.is_binary:
-            state = {
-                ATTR_MEAN: self.mean,
+            return {
+                ATTR_SAMPLING_SIZE: self._sampling_size,
                 ATTR_COUNT: self.count,
-                ATTR_MAX_VALUE: self.max,
+                ATTR_MEAN: self.mean,
                 ATTR_MEDIAN: self.median,
-                ATTR_MIN_VALUE: self.min,
-                ATTR_SAMPLING_SIZE: self._sampling_size,
                 ATTR_STANDARD_DEVIATION: self.stdev,
-                ATTR_TOTAL: self.total,
                 ATTR_VARIANCE: self.variance,
+                ATTR_TOTAL: self.total,
+                ATTR_MIN_VALUE: self.min,
+                ATTR_MAX_VALUE: self.max,
+                ATTR_MIN_AGE: self.min_age,
+                ATTR_MAX_AGE: self.max_age,
```

Comment:
If the configuration option is not set to use `max_age` then `max_age`/`min_age` should not be exposed. The timestamps are not useful because it's the timestamp for the first value and the last sensor update will write a new timestamp. And this timestamp is the same as the one stored in the entity state.

Reply:
To display the timestamp of the oldest/newest sensor value is the single use case for this, as this information is not available in e.g. a template sensor to calculate the time range, etc.

My first intention only was to always export the `max_age`/`min_age` and calculate change_rate in a template sensor but I then added this as a feature directly. I see no difference in always exporting these values and only exporting them when `max_age` is specified. They always have the same meaning. On the other hand, exporting `max_age`/`min_age` when `max_age` is used seems the less useful case as `max_age` (the config option) implies a approximation for `max_age` (and `min_age` for `now()`) where using only sampling_size doesn't give any approximation or even a rough scale for any timestamp.